cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate/FreePlanAwardedCourseCertificate.js

```diff
@@ -70,8 +70,14 @@ Then('the user should see the call to action "Download certificate" for awarded
 });
 
 Then('the certificate should be downloaded successfully', () => {
+  const filePath = './cypress/downloads/user_Course_Activity_Test_2_2024.pdf';
   cy.contains('Download certificate').click();
-  cy.readFile('./cypress/downloads/user_Course_Activity_Test_2_2024.pdf').should('exist');
+  cy.readFile(filePath).should('exist');
+
+    // delete created files after run to avoid false positives
+    cy.task('deleteFile', filePath).then(result => {
+      expect(result).to.be.null;
+    });
```

Comment:
Please use a cucumber hook instead, I would recommend using `Before/BeforeAll`, because it allow us to be in a safe fresh start even if the test fails.

https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/cucumber-basics.md#hooks

https://www.reddit.com/r/QualityAssurance/comments/pc96o6/e2e_testing_using_cypress_how_to_clean_up_state/

Reply:
To be less tightly coupled to the name of the file, please, remove everything in the downloads directory

Reply:
The same comment applies to the other files, in general, it's not a good practice to do clean-up inside the test assert.